pkg/controller/azdrivernode.go

```diff
@@ -124,9 +135,14 @@ func NewAzDriverNodeController(mgr manager.Manager, controllerSharedState *Share
 		return nil, err
 	}
 
-	// Predicate to only reconcile deleted nodes
+	// Predicate to only reconcile deleted nodes.
 	p := predicate.Funcs{
+		// For create node event, add the new node to nodeDiskAvailabilityMap
 		CreateFunc: func(e event.CreateEvent) bool {
+			diskCount, err := azureutils.GetNodeMaxDiskCountWithLabels(e.Object.GetLabels())
+			if err == nil {
+				reconciler.nodeDiskAvailabilityMap.Store(e.Object.GetName(), diskCount)
```

Comment:
Predicates should not have side-effects. The Create predicate should simply return true and the Reconcile function should add the corresponding entry when it observes a node for which there is no current entry.

Reply:
Done